File: 2002/101 VB.NET Samples/VB.NET - Data Access - Retreive and Process data with a SQL Data Reader/Category.vb
' The Category class is a custom type that is used for databinding to a ComboBox.
' Notice the use of public properties instead of merely public fields. You might
' think you could use the following construct:
'
'   Public Class Category
'       Public ID As Integer
'       Public Name As String

'       Sub New(ByVal intID As Integer, ByVal strName As String)
'           ID = intID
'           Name = strName
'       End Sub
'   End Class
'
' This will, however, throw a runtime InvalidArgumentException stating that it 
' cannot bind to the new DisplayMember. The Property does not have to be 
' ReadOnly.
Class Category
    Dim _ID As Integer
    Dim _Name As String

    Sub New(ByVal intID As Integer, ByVal strName As String)
        _ID = intID
        _Name = strName
    End Sub

    Public ReadOnly Property ID() As Integer
        Get
            Return _ID
        End Get
    End Property

    Public ReadOnly Property Name() As String
        Get
            Return _Name
        End Get
    End Property
End Class

File: 2003/Code Samples/Data Access Samples/Data Access - Retreive and Process data with a SQL Data Reader/Category.cs
// The Category class is a custom type that is used for databinding to a ComboBox.
// Notice the use of public properties instead of merely public fields. You might
// think you could use the following construct:
//
//   public class Category
//       public ID int
//       public Name string
//       void new(intID int, strName string)
//           ID = intID
//           Name = strName
//       }
//   }
//
// This will, however, throw a runtime InvalidArgumentException stating that it 
// can! bind to the new DisplayMember. The DATA_TYPE_HERE does not have to be 
// ReadOnly.

class Category
{

    int _ID;

    string _Name;

    public Category(int intID, string strName)
	{
        _ID = intID;
        _Name = strName;
    }

    public int ID
	{

        get 
		{
            return _ID;
        }

    }

    public string Name
	{

        get 
		{
            return _Name;
        }

    }

}

